src/services/quotes/quote-sources/dodo-quote-source.ts

```diff
@@ -0,0 +1,93 @@
+import { Chains } from '@chains';
+import { AlwaysValidConfigAndContextSource } from './base/always-valid-source';
+import { QuoteParams, QuoteSourceMetadata, SourceQuoteResponse } from './types';
+import qs from 'qs';
+import { addQuoteSlippage, calculateAllowanceTarget, failed } from './utils';
+import { parseUnits } from 'viem';
+
+const DODO_DEX_METADATA: QuoteSourceMetadata<DodoDexSupport> = {
+  name: 'DODO Dex',
+  supports: {
+    chains: [
+      Chains.ETHEREUM.chainId,
+      Chains.BNB_CHAIN.chainId,
+      Chains.ARBITRUM.chainId,
+      Chains.POLYGON.chainId,
+      Chains.AURORA.chainId,
+      Chains.AVALANCHE.chainId,
+      Chains.BASE.chainId,
+      Chains.BOBA.chainId,
+      Chains.BASE_GOERLI.chainId,
+      Chains.HECO.chainId,
+      Chains.LINEA.chainId,
+      Chains.MOONRIVER.chainId,
+      Chains.OKC.chainId,
+      Chains.OPTIMISM.chainId,
+    ],
+    swapAndTransfer: false,
+    buyOrders: false,
+  },
+  logoURI: '',
+};
+type DodoDexConfig = { apiKey: string };
+type DodoDexSupport = { buyOrders: false; swapAndTransfer: false };
+export class DodoDexQuoteSource extends AlwaysValidConfigAndContextSource<DodoDexSupport, DodoDexConfig> {
```

Comment:
Is the `apiKey` necessary, or optional? If it is necessary, then it wouldn't be always valid